CONTRIBUTING.md

```diff
@@ -25,13 +25,22 @@ aiconfig/
 ├─ editor/
 ├─ aiconfig-docs/
 ├─ python/
-│   ├── model-parsers/
 ├─ typescript/
-│   ├── model-parsers/
 ├─ extensions/
 ```
 
-> *To Do: Add information on library structure and contribution opportunities*
+1. **{ Model Parsers }**: Model Parsers manage the conection to foundation models. By default, we provide the OpenAI models and Google PaLM models. The model parsers are defined in both `python` and `typescript`. 
+   
+    Instructions for contributing a Model Parser:
+    1. Add a folder under `aiconfig/extensions`
+    2. Define the requirements. For `python`, you'll need to set up `pyproject.toml`. For `typescript`, you'll need a `package.json`.
+   > **Naming convention for the extension**: *aiconfig-extension-NAME_OF_MODEL_PARSER*.
+    3. Extend from `ModelParser` class, and implement `serialize`, `deserialize` and `run` functions.
+    4. We highly recommend adding a cookbook under `aiconfig/cookbooks` for a guide on using your model parser.
```

Comment:
Should this be a requirement?

Reply:
Good question. I was thinking this might drastically reduce the model parsers, but your call